### PR TITLE
Refactor PMIx_Get fixing `PMIX_store_internal` case

### PR DESCRIPTION
**Pull request by karasevb**
Signed-off-by: Boris Karasev <karasev.b@gmail.com>
(cherry picked from commit 496826a)

### CODE REVIEW

**Comment by karasevb** [comment]
fixes #316

File: src/client/pmix_client_get.c [hunk 1]
```diff
@@ -535,118 +536,108 @@ static void _getnbfn(int fd, short flags, void *cbdata)
...
-     * ourselves during startup */
+        /* it could still be in the job-data table, only stored under its own
+         * rank and not WILDCARD - e.g., this is true of data returned about
+         * ourselves during startup */
 #if defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1)
-    if (PMIX_SUCCESS == (rc = pmix_dstore_fetch(nptr->nspace, cb->rank, cb->key, &val))) {
+        rc = pmix_dstore_fetch(nptr->nspace, cb->rank, cb->key, &val);
 #else
-    if (PMIX_SUCCESS == (rc = pmix_hash_fetch(&nptr->internal, cb->rank, cb->key, &val))) {
+        rc = pmix_hash_fetch(&nptr->internal, cb->rank, cb->key, &val);
```
Comment (artpol84) [inline src/client/pmix_client_get.c]: Already done in line 540

Comment (artpol84) [inline src/client/pmix_client_get.c]:
```C
rc = PMIX_ERR_NOT_FOUND
```

File: src/client/pmix_client_get.c [hunk 2]
```diff
@@ -535,118 +536,108 @@ static void _getnbfn(int fd, short flags, void *cbdata)
...
 #endif
-        /* found it - we are in an event, so we can
-         * just execute the callback */
-        cb->value_cbfunc(rc, val, cb->cbdata);
-        /* cleanup */
-        if (NULL != val) {
-            PMIX_VALUE_RELEASE(val);
+        if (PMIX_SUCCESS == rc) {
```
Comment (artpol84) [inline src/client/pmix_client_get.c]: don't need this

File: src/client/pmix_client_get.c [hunk 3]
```diff
@@ -535,118 +536,108 @@ static void _getnbfn(int fd, short flags, void *cbdata)
...
     }
 
-    /* not finding it is not an error - it could be in the
-     * modex hash table, so check it */
+    /* otherwise, the data must be something they "put" */
 #if defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1)
-    rc = pmix_dstore_fetch(nptr->nspace, cb->rank, cb->key, &val);
+    rc = PMIX_ERR_PROC_ENTRY_NOT_FOUND;
+    /* if rank is undefined - check local table first */
+    if ( my_nspace && (PMIX_RANK_UNDEF == cb->rank)){
```
Comment (artpol84) [inline src/client/pmix_client_get.c]: Please, double-check how v1.2 works with `PMIX_RANK_UNDEF`?

Comment (artpol84) [inline src/client/pmix_client_get.c]: Check this against PMI1/PMI2 clients.

Comment (artpol84) [inline src/client/pmix_client_get.c]: dstore_fetch supports UNDEF rank => this extra check is not needed

**Comment by jjhursey** [comment]
What's the status of this PR?

**Comment by karasevb** [comment]
It was automatically closed when I deleted a branch mistakenly.

**Pull request closed without merging**